controller/models/Instance.go

```diff
@@ -13,7 +13,83 @@
 // limitations under the License.
 package models
 
+import (
+	"bytes"
+	"encoding/json"
+)
+
 // Instance is the external view of instances
-type Instance struct {
+type InstanceView struct {
 	Address string `json:"address"`
 }
+
+// InstanceState define instance state
+type InstanceState int
+
+const (
+	// Active indicates an instance is healthy
+	Active InstanceState = iota
+	// InActive indicates an instance is not healthy
+	InActive
+)
+
+// MarshalJSON is customized
+func (i InstanceState) MarshalJSON() ([]byte, error) {
+	stateString := "active"
+	if i != Active {
+		stateString = "inactive"
+	}
+	buffer := bytes.NewBufferString(`""`)
+	buffer.WriteString(stateString)
+	buffer.WriteString(`""`)
+	return buffer.Bytes(), nil
+}
+
+// UnmarshalJSON is customized
+func (i InstanceState) UnmarshalJSON(data []byte) error {
+	i = InActive
+	if string(data) == "active" {
+		i = Active
+	}
+	return nil
+}
+
+// Instance models an aresdb instance
+type Instance struct {
+	Name  string        `json:"name"`
```

Comment:
what is the difference between name and host here? or add some more comments

Reply:
Name is an internal ID known by controller, host name is the physical host name. they are the same for now, but could be different in the future.